extensions/azure/storage/AzureDataLakeStorage.cpp

```diff
@@ -20,11 +20,43 @@
 
 #include "AzureDataLakeStorage.h"
 
+#include <regex>
+#include <string_view>
+
 #include "AzureDataLakeStorageClient.h"
 #include "io/StreamPipe.h"
+#include "utils/file/FileUtils.h"
+#include "utils/StringUtils.h"
+#include "utils/gsl.h"
+#include "utils/GeneralUtils.h"
 
 namespace org::apache::nifi::minifi::azure::storage {
 
+namespace {
+bool matchesPathFilter(std::string_view base_directory, const std::string& path_filter, std::string path) {
```

Comment:
we could propagate the `std::string -> std::regex` conversion up the chain (make it a responsibility of `onSchedule`) and accept an `std::optional<std::regex>` here, in case of an invalid regex expression we would get immediate feedback in `onSchedule` instead of on every trigger